pkg/proxy/proxy.go

```diff
@@ -54,7 +54,9 @@ func Setup(obj *unstructured.Unstructured) error {
 // path... -> Pod, DaemonSet, BuildConfig, etc.
 func SetupDaemonSet(obj *unstructured.Unstructured) error {
 	containers, found, err := unstructured.NestedSlice(obj.Object, "spec", "template", "spec", "containers")
-	exit.OnErrorOrNotFound(found, err)
+	if err != nil || !found {
+		return err
```

Comment:
Distinguish between error and not found? If its not found err is nil and this is returned as if it worked ok.

Reply:
Fixed.